src/SelfService/Application/CapabilityApplicationService.cs

```diff
@@ -266,4 +267,47 @@ public async Task<string> GetJsonMetadata(CapabilityId id)
     {
         return await _capabilityRepository.GetJsonMetadata(id);
     }
+
+    public async Task<bool> DoesOnlyModifyRequiredProperties(string jsonMetadata, CapabilityId capabilityId)
+    {
+        var capability = await _capabilityRepository.Get(capabilityId);
+
+        var previousMetadata = capability.JsonMetadata;
+        var jsonSchema = await _selfServiceJsonSchemaService.GetLatestSchema(SelfServiceJsonSchemaObjectId.Capability);
+        if (jsonSchema == null)
+        {
+            return false;
+        }
+
+        var jsonSchemaAsObject = JsonNode.Parse(jsonSchema.Schema)?.AsObject()!;
+        var previousMetadataAsObject = JsonNode.Parse(previousMetadata)?.AsObject()!;
+        var jsonMetadataAsObject = JsonNode.Parse(jsonMetadata)?.AsObject()!;
+        var requiredProperties = jsonSchemaAsObject.TryGetPropertyValue("required", out var requiredPropertiesNode);
+
+        var requiredPropertiesKeys = requiredPropertiesNode?.AsArray().Select(x => x!.ToString()).ToList()!;
+
+        // Check if new metadata contain the properties from the previous metadata and if not check if its required
+        foreach (var c in previousMetadataAsObject)
+        {
+            if (jsonMetadataAsObject.ContainsKey(c.Key))
+                continue;
+            if (!requiredProperties)
+                return false;
+            if (!requiredPropertiesNode!.AsArray().Contains(c.Key))
+                return false;
+        }
+
+        // Check if new json metadata contain new properties that are not in the previous metadata and if not check if its required
+        foreach (var c in jsonMetadataAsObject)
+        {
+            if (previousMetadataAsObject.ContainsKey(c.Key))
+                continue;
+            if (!requiredProperties)
```

Comment:
This does not seem dependant on the loop variable at all, so could this not be moved out of the loop entirely?